truss/templates/control/control/helpers/inference_server_controller.py

```diff
@@ -113,3 +114,22 @@ def _check_and_recover_inference_server(self):
             with self._lock:
                 self._process_controller.check_and_recover_inference_server()
             time.sleep(INFERENCE_SERVER_CHECK_INTERVAL_SECS)
+
+
+def _patch_sort_key_fn(patch: Patch) -> int:
+    # System packages need to be applied before python requirements as they
+    # might need them.
+    # System packages or python packages are more likely to fail to apply then
+    # model code patches, so it's better to apply model code patches last to
+    # avoid ending up with partially applied patches which currently triggers
+    # full deploy.
+    if patch.type == PatchType.SYSTEM_PACKAGE:
```

Comment:
Is this something that we can get from the `PatchType` enum instead of having to maintain this ordering mapping separately and we can just use the value of the enum?

Reply:
We have to store the type in the json anyway to know how to parse the body, so the patch type is required. Here we could check the patch type or the type of the enum, either should be fine, this seems more direct.